Bank_Account.rb

```diff
@@ -35,23 +35,23 @@ def add_owner(owner) #adds owner from an existing owner, rather than making a ne
       puts "\n#{@owner.first_name} is now the owner of account \##{@account_id}.\n"
     end
 
-    def withdraw(withdrawal_amount)
-      # print "\nEnter value to be withdrawn: "
-      # withdrawal_amount = gets.chomp.to_i
-      if withdrawal_amount > @balance
-        puts "\nWithdrawal denied."
-        puts "Balance: #{@balance}\n"
-        return @balance
+    def reject_withdrawal
+      puts "\nWithdrawal denied."
+      puts "Balance: #{@balance}\n"
+      return @balance
+    end
+
+    def withdraw(withdrawal_amount, withdrawalcap = @balance)
+      if (withdrawal_amount + @withdrawalfee) > withdrawalcap
+        reject_withdrawal
       else
-        @balance = @balance - withdrawal_amount
+        @balance -= withdrawal_amount + @withdrawalfee
         puts "\nBalance after withdrawal: #{@balance}\n"
         return @balance
       end
     end
 
     def deposit(deposit_amount)
-      # print "\nEnter value to be deposited: "
-      # deposit_amount = gets.chomp.to_i
       @balance = @balance + deposit_amount
       puts "\nBalance after deposit: #{@balance}\n"
       return @balance
```

Comment:
In the `generate_accounts` method below, I would recommend using a more explicit variable name other than `a`